core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```diff
@@ -724,19 +724,29 @@ export class BaseUser {
               path.basename(dirName)
             )
           : path.join(testPath, dirName, 'diff-snapshots'),
+        storeReceivedOnFailure: true,
```

Comment:
I think some comments to explain what's happening in these lines might be helpful.